batch7crud-roll8/src/main/java/com/lftechnology/batch7crud/controller/EmployeesController.java

```diff
@@ -0,0 +1,206 @@
+      response.sendRedirect(request.getContextPath() + UrlConstants.EMPLOYEE_ROUTE);
+    } catch (Exception e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      errorPage(request, response, e.getMessage());
+    }
+  }
+
+  private void editProcess(HttpServletRequest request, HttpServletResponse response) {
+    try{
+      int id = parameterValueAsInt(request, 2);
+      Employee employee = setEmployeeAttributes(request);
+      employee.setId(id);
+      employeeService.edit(employee);
+      response.sendRedirect(request.getContextPath() + UrlConstants.EMPLOYEE_ROUTE);
+    }
+    catch (Exception e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      errorPage(request, response, e.getMessage());
+    }
+  }
+
+  private void deleteProcess(HttpServletRequest request, HttpServletResponse response) {
+    try {
+      int id = parameterValueAsInt(request, 2);
+      employeeService.delete(id);
+      response.sendRedirect(request.getContextPath() + UrlConstants.EMPLOYEE_ROUTE);
+    }
+    catch (Exception e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      errorPage(request, response, e.getMessage());
+    }
+  }
+
+  private void fetch(HttpServletRequest request, HttpServletResponse response) {
+    try {
+      Integer pageLimit = AppConstants.PAGE_LIMIT;
+      Integer pageNo = getCurrentPage(request);
+      Integer offset = (pageNo-1)*pageLimit;
+
+      List<Employee> employeeList = employeeService.fetch(pageLimit, offset);
+      Integer employeeCount = employeeService.count();
+      request.setAttribute(AttributeConstants.EMPLOYEE_LIST, employeeList);
+      request.setAttribute(AttributeConstants.EMPLOYEE_COUNT, employeeCount);
+      request.setAttribute(AttributeConstants.PAGE_NO, pageNo);
+      if ((employeeCount%pageLimit) == 0) {
+        request.setAttribute(AttributeConstants.LAST_PAGE_NO, employeeCount / pageLimit);
+      }
+      else {
+        request.setAttribute(AttributeConstants.LAST_PAGE_NO, (employeeCount / pageLimit) + 1);
+      }
+      request.getServletContext().getRequestDispatcher(request.getContextPath() + UrlConstants.EMPLOYEE_LIST_PAGE).forward(request, response);
+    }
+    catch (Exception e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      errorPage(request, response, e.getMessage());
+    }
+  }
+
+  private void view(HttpServletRequest request, HttpServletResponse response) {
+    try {
+      int id = parameterValueAsInt(request, 2);
+      Employee employee = employeeService.fetchById(id);
+      if (employee != null) {
+        request.setAttribute(AttributeConstants.EMPLOYEE, employee);
+        request.getRequestDispatcher(request.getContextPath() + UrlConstants.EMPLOYEE_VIEW_PAGE).forward(request, response);
+      }
+      else {
+        pageNotFound(request, response);
+      }
+    }
+    catch (NumberFormatException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      pageNotFound(request, response);
```

Comment:
you don't need to mention pageNotFound  here after you map 404 which is responsible for page not found http status. You just need to set the status code in your response and it will be automatically handle thru web.xml file.